modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectField/Tabs/BasicInfo/AggregationFilterContainer.tsx

```diff
@@ -0,0 +1,560 @@
+					const aggregationFilter: AggregationFilters = {
+						fieldLabel: getLocalizableLabel(
+							creationLanguageId2 as Locale,
+							objectField.label,
+							objectField.name
+						),
+						filterBy: parsedFilter.filterBy,
+						filterType,
+						label: objectField.label,
+						objectFieldBusinessType: objectField.businessType,
+						objectFieldName: objectField.name,
+						value:
+							objectField.businessType === 'Integer' ||
+							objectField.businessType === 'LongInteger'
+								? (parsedFilter.json as {
+										[key: string]: string;
+								  })[filterType]
+								: undefined,
+					};
+
+					if (
+						objectField.businessType === 'Date' &&
+						parsedFilter.filterType === 'range'
+					) {
+						const dateRangeFilterValues = parsedFilter.json as ObjectFieldDateRangeFilterSettings;
+
+						const aggregationFilterDateRangeValues: LabelValueObject[] = [
+							{
+								label: dateRangeFilterValues['ge'],
+								value: 'ge',
+							},
+							{
+								label: dateRangeFilterValues['le'],
+								value: 'le',
+							},
+						];
+
+						const dateRangeAggregationFilter: AggregationFilters = {
+							...aggregationFilter,
+							valueList: aggregationFilterDateRangeValues,
+						};
+
+						return dateRangeAggregationFilter;
+					}
+
+					if (objectField.businessType === 'Picklist') {
+						const picklistFilterValues = getPicklistFilterJSONValues(
+							filterType,
+							parsedFilter
+						) as string[];
+
+						const picklistValueList: LabelValueObject[] = picklistFilterValues.map(
+							(picklistFilterValue) => {
+								return {
+									checked: true,
+									label: picklistFilterValue,
+									value: picklistFilterValue,
+								};
+							}
+						);
+
+						const picklistAggregationFilter: AggregationFilters = {
+							...aggregationFilter,
+							valueList: picklistValueList,
+						};
+
+						return picklistAggregationFilter;
+					}
+
+					if (objectField.name === 'status') {
+						const statusFilterValues = getPicklistFilterJSONValues(
+							filterType,
+							parsedFilter
+						) as number[];
+
+						const workflowStatusValueList = statusFilterValues.map(
+							(statusValue) => {
+								const currentStatus = workflowStatusJSONArray.find(
+									(workflowStatus) =>
+										Number(workflowStatus.value) ===
+										statusValue
+								);
+
+								return {
+									label: currentStatus?.label,
+									value: currentStatus?.label,
+								};
+							}
+						);
+
+						const statusAggregationFilter: AggregationFilters = {
+							...aggregationFilter,
+							valueList: workflowStatusValueList as LabelValueObject[],
+						};
+
+						return statusAggregationFilter;
+					}
+
+					return aggregationFilter;
+				}
+			});
+
+			setAggregationFilters(
+				newAggregationFilters as AggregationFilters[]
+			);
+		}
+
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [objectFields]);
+
+	const validateFilters = ({
```

Comment:
Hey @alinedoleron, as we are passing this function to `ModalAddFilter`, to improve performance and avoid re-rendering, you can wrap this function by [useCalback hook](https://beta.reactjs.org/reference/react/useCallback#usage) which is useful precisely in these cases